src/common/utils/validator.js

```diff
@@ -217,18 +217,31 @@ function currentOrFutureDate(dObj) {
   return false;
 }
 
-// function dateTooFarInFuture(dObj){
-//   if (dObj === null || dObj === undefined) return false;
+function dateNotTooFarInFuture(dObj) {
+  if (dObj === null || dObj === undefined) return false;
+
+  const now = new Date();
+  var nextMonth = new Date(now.getFullYear(), now.getMonth() + 1, now.getDate());
+
+
+  let providedDate;
+
+  if (dObj instanceof Date) {
+    providedDate = dObj;
+  } else {
+
+    if (!(numericDateElements(dObj)
+      && validDay(dObj.d, dObj.m, dObj.y)
+      && validMonth(dObj.m)
+      && validYear(dObj.y))) {
+      return false;
```

Comment:
Can we verify this check, please - currently this is returning false if it is not a numericvalue but a valid date/month/day

Reply:
It is good to have the check but we just need to double check if the condition is correct.

Reply:
have confirmed with Anusha that condition is correct